src/Internal/Requirements/RequirementValidator.php

```diff
@@ -39,6 +39,14 @@ public static function instance(): RequirementValidator {
 	 * @param RuntimeException $e
 	 */
 	protected function add_admin_notice( RuntimeException $e ) {
+		// Check if the plugin is active. If not, display error message and terminate.
+		if ( ! is_plugin_active( 'woocommerce-google-listings-and-ads/woocommerce-google-listings-and-ads.php' ) ) {
```

Comment:
Another point about `is_plugin_active`, that function is only available in certain circumstances (when plugin is activating or after admin_init hook). So if I have the plugin already activated and then I simulate one of the scenarios where it's missing a requirement it gives a fatal error that the function is undefined.

Maybe something like this is needed:
```php
if ( ! function_exists( 'is_plugin_active' ) ) {
	require_once ABSPATH . '/wp-admin/includes/plugin.php';
}

Reply:
Thanks @mikkamp, I have added this check.